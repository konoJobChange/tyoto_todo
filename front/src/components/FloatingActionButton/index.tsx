import { useCallback } from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      '& > *': {
        margin: theme.spacing(1),
      },
    },
    fab: {
      position: 'fixed',
      right: '1rem',
      bottom: '1rem',
    },
  }),
);

export default function FloatingActionButton() {
  const classes = useStyles();

  const handleCreateTodo = useCallback(() => {
    console.log('create');
  }, []);

  return (
    <div className={classes.root}>
      <Fab
        color="primary"
        aria-label="create todo"
        className={classes.fab}
        onClick={handleCreateTodo}
      >
        <AddIcon />
      </Fab>
    </div>
  );
}
