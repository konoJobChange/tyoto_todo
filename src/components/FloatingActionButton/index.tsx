import { useCallback, useState } from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import InputDialog from 'src/components/InputDialog';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      '& > *': {
        margin: theme.spacing(1),
      },
    },
    fab: {
      position: 'fixed',
      right: '1rem',
      bottom: '1rem',
    },
  }),
);

export default function FloatingActionButton() {
  const classes = useStyles();
  const [open, setOpen] = useState(false);

  const handleOpen = useCallback(() => {
    setOpen(true);
  }, []);

  const handleClose = useCallback<() => void>(() => {
    setOpen(false);
  }, []);

  return (
    <div className={classes.root}>
      <Fab
        color="primary"
        aria-label="create todo"
        className={classes.fab}
        onClick={handleOpen}
      >
        <AddIcon />
      </Fab>
      <InputDialog open={open} handleClose={handleClose} />
    </div>
  );
}
