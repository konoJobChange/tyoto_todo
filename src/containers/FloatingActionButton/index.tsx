import { useCallback, useState } from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import InputDialog from 'src/components/InputDialog';
import { useAuth } from 'src/modules/hooks/useAuth';
import { useUpdate } from 'src/modules/hooks/useTodos';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      '& > *': {
        margin: theme.spacing(1),
      },
    },
    fab: {
      position: 'fixed',
      right: '1rem',
      bottom: '1rem',
    },
  }),
);

export default function FloatingActionButton() {
  const classes = useStyles();
  const [open, setOpen] = useState(false);
  const { user } = useAuth();

  const handleOpen = useCallback(() => {
    setOpen(true);
  }, []);

  const handleClose = useCallback<() => void>(() => {
    setOpen(false);
  }, []);

  const handleCreate = useCallback(
    async (title: string, detail: string) => {
      await useUpdate(title, detail, user);
      handleClose();
    },
    [handleClose, user],
  );

  return (
    <div className={classes.root}>
      <Fab
        color="primary"
        aria-label="create todo"
        className={classes.fab}
        onClick={handleOpen}
      >
        <AddIcon />
      </Fab>
      <InputDialog
        open={open}
        handleClose={handleClose}
        handleCreate={handleCreate}
      />
    </div>
  );
}
