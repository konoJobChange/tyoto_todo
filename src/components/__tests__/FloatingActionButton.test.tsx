import { shallow } from 'enzyme';
import FloatingActionButton from 'src/components/FloatingActionButton';

test('CheckboxWithLabel changes the text after click', () => {
  const button = shallow(<FloatingActionButton />);

  // Interaction demo
  expect(button.text()).toEqual('Off');
  button.find('input').simulate('change');
  expect(button.text()).toEqual('On');

  // Snapshot demo
  expect(button).toMatchSnapshot();
});
