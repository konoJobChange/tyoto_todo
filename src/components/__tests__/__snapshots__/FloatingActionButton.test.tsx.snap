// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDialogをチェックするーーーー 1`] = `
<div>
  <WithStyles(ForwardRef(Dialog))
    aria-labelledby="form-dialog-title"
    open={true}
  >
    <WithStyles(ForwardRef(DialogTitle))
      id="form-dialog-title"
    >
      Todo
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(DialogContent))>
      <WithStyles(ForwardRef(DialogContentText))>
        To create your Todo.
        <br />
        This is only your tyoto_todo!!
      </WithStyles(ForwardRef(DialogContentText))>
      <WithStyles(ForwardRef(TextField))
        autoFocus={true}
        fullWidth={true}
        id="title"
        label="Title"
        margin="dense"
        onChange={[Function]}
        required={true}
        type="text"
        value="ブス"
      />
      <WithStyles(ForwardRef(TextField))
        fullWidth={true}
        id="detail"
        label="Detail"
        margin="dense"
        multiline={true}
        onChange={[Function]}
        rowsMax="3"
        value="ブスには価値がない"
      />
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogActions))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [
                "ブス",
                "ブスには価値がない",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      >
        Cancel
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        data-testId="create"
        onClick={[Function]}
      >
        Create
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(DialogActions))>
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
